### PR TITLE
ccache: add a compiler signature to the cache in order to handle comp…

### PR DESCRIPTION
**Pull request by jerome-benoit**
…iler upgrade flawlessly

Signed-off-by: Jérôme Benoit jerome.benoit@piment-noir.org

### CODE REVIEW

**Comment by stefansaraev** [comment]
well... get_pkg_dir is missing.

you should also clean up gcc package (ccache trashing logic by stamp)

**Comment by stefansaraev** [comment]
before this goes in, I suggest reworking ccache as it is done in openelec, and moivng ccache as per-projcet cache dir.

**Comment by jerome-benoit** [comment]
Feel free to add the relevant commits to this branch under the Piment Noir github.

File: config/functions [hunk 1]
```diff
@@ -137,6 +137,37 @@ get_build_dir() {
   fi
 }
 
+get_pkg_dir() {
+  PKG_EXTRA_FOLDERS="$HOME/.libreelec/projects/$PROJECT/packages \
```
Comment (lrusak) [inline config/functions]: nope

File: config/functions [hunk 2]
```diff
@@ -137,6 +137,37 @@ get_build_dir() {
...
+                     $HOME/.libreelec/packages \
```
Comment (lrusak) [inline config/functions]: also nope

**Comment by MilhouseVH** [comment]
There's no need to implement the `get_pkg_dir() function (which, btw, will only work for packages in the `packages` folder, and not for packages in the `projects` folder).

If you replace `get_pkg_version()` with the following, you can lose `get_pkg_dir()` entirely:
```
+get_pkg_version() {
+  if [ ! -z $1 ] ; then
+    . $ROOT/config/options $1
+    echo ${PKG_VERSION}
+  fi
+}
```

**Comment by chewitt** [comment]
This has been gathering dust. If it is still something desirable? (It's not my area of expertise to make comments upon so I'd defer to @MilhouseVH @stefansaraev) it needs rebasing.

**Comment by chewitt** [comment]
I'm closing this PR due to lack of comments (or rebase)

**Pull request closed without merging**